src/trio/_threads.py

```diff
@@ -351,6 +351,8 @@ async def to_thread_run_sync(
     if limiter is None:
         limiter = current_default_thread_limiter()
 
+    # TODO: cancel_register can probably be a single element tuple for typing reasons
```

Comment:
This helps in stopping people from indexing above 0, but it breaks a bunch of current code that manipulates the content of the list. Ideally it'd be some minimalist wrapper class.

Reply:
would a deque with maxlen=1 help?

Reply:
I wouldn't expect typing to take the maxlen into account